### ISSUE TITLE
Collab Offer

### ISSUE THREAD

**Issue by AlanOC91** [chinhodado/ygodb#7]
Hi there,

Would you be interested in assisting us with our [Online Yu-Gi-Oh! Database](https://db.ygoprodeck.com/). 

This is a one time piece of work we need done. We are struggling to parse Card Rulings from Yugipedia/Yu-Gi-Oh! Wikia and by the looks of things, you have done an excellent job in parsing them.

I would be willing to compensate you for this work.

Thanks!

**Comment by HoangDynasty** [comment]
I haven't parsed anything, what are you talking about?

**Comment by AlanOC91** [comment]
I've been attempting to unsuccessfully parse card rulings from Yugipedia unsuccessfully (using this [HTML parser](https://github.com/paquettg/php-html-parser)). 
Forgive my phrasing, parse is the wrong phrase for what you are doing with this.

We basically just need a way to pull card rulings for each card into a database and store them for offline use. You seem like the perfect person to accomplish this since you have knowledge of acquiring Card Rulings already.

Would you be interested?

**Comment by GX4600** [comment]
What exactly would you need me to do ?
Sent from Yahoo Mail on Android 
 
  On Wed, Jul 10, 2019 at 6:58 AM, YGOPRODeck<notifications@github.com> wrote:   
I've been attempting to unsuccessfully parse card rulings from Yugipedia unsuccessfully (using this HTML parser).
Forgive my phrasing, parse is the wrong phrase for what you are doing with this.

We basically just need a way to pull card rulings for each card into a database and store them for offline use. You seem like the perfect person to accomplish this since you have knowledge of acquiring Card Rulings already.

Would you be interested?

**Comment by AlanOC91** [comment]
Basically I'd need pretty much a two column csv file that contains card ID + Rulings. If card ID is not possible then name would work. We only need the OCG Rulings listed.

For example Dark Magician of Chaos would be:

| ID (or name) | Ruling |
| --- | --- |
| 40737112 | The effect "During the End Phase, if this card was Normal or Special Summoned this turn: You can target 1 Spell Card in your Graveyard; add that target to your hand." is a Trigger Effect. (It does not start a Chain Link when this card is Normal or Special Summoned. It is activated during the End Phase. It targets 1 Spell Card in your Graveyard.) |
| 40737112 | During the turn "Dark Magician of Chaos" was Normal or Special Summoned, this effect cannot be activated if it leaves the field before the End Phase, for example, by being destroyed. |
| 40737112 | The effect "If this card destroys an opponent's monster by battle, after damage calculation: Banish that opponent's monster." is a Trigger Effect. (It does not target. It is a mandatory effect that activates when this card destroys an opponent's monster by battle.) |
| 40737112 | If "Dark Magician of Chaos" battles a monster with the same ATK and both monsters are destroyed by battle, this effect still activates, and the opponent's monster is banished. |
| 40737112 | The effect "If this face-up card would leave the field, banish it instead." is not treated as an Ignition, Trigger, Quick or Continuous Effect. |

And so on for each card. I can provide a csv file that contains every card Name/ID for you.

**Comment by GX4600** [comment]
My only concern is that I have rated mobile internet connection that gets throttled after 50GB of data use. I would have to wait an entire month before it resets again. 
I could jump on a wifi network but at the moment I find myself recovering from a few recent medical procedures.
Have you tried Yu-Gi-Oh! Card Database - YGOPRODECK  
|  
|   
|   
|   |    |

   |

  |
|  
|   |  
Yu-Gi-Oh! Card Database - YGOPRODECK
 
Find every Yu-Gi-Oh! card including both TCG cards and OCG from every set in the game, easily sorted by class, type, level, and more.
  |   |

  |

  |

  
They seem to be very active in organizing yugioh card information. 

Sent from Yahoo Mail on Android 
 
  On Wed, Jul 10, 2019 at 7:14 AM, YGOPRODeck<notifications@github.com> wrote:   
Basically I'd need pretty much a two column csv file that contains card ID + Rulings. If card ID is not possible then name would work.

For example Dark Magician of Chaos would be:

| ID (or name) | Ruling |
| 40737112 | The effect "During the End Phase, if this card was Normal or Special Summoned this turn: You can target 1 Spell Card in your Graveyard; add that target to your hand." is a Trigger Effect. (It does not start a Chain Link when this card is Normal or Special Summoned. It is activated during the End Phase. It targets 1 Spell Card in your Graveyard.) |
| 40737112 | During the turn "Dark Magician of Chaos" was Normal or Special Summoned, this effect cannot be activated if it leaves the field before the End Phase, for example, by being destroyed. |
| 40737112 | The effect "If this card destroys an opponent's monster by battle, after damage calculation: Banish that opponent's monster." is a Trigger Effect. (It does not target. It is a mandatory effect that activates when this card destroys an opponent's monster by battle.) |
| 40737112 | If "Dark Magician of Chaos" battles a monster with the same ATK and both monsters are destroyed by battle, this effect still activates, and the opponent's monster is banished. |
| 40737112 | The effect "If this face-up card would leave the field, banish it instead." is not treated as an Ignition, Trigger, Quick or Continuous Effect. |


And so on for each card. I can provide a csv file that contains every card Name/ID for you.

**Comment by AlanOC91** [comment]
Yes the Yu-Gi-Oh! Card Database is the site I own and run :P Thank you!  I have personally found Card Rulings to be the absolute hardest piece of information to programmatically obtain which is why I am reaching out to someone who has more experience acquiring Card Rulings. I would love to add it to my card database.

I completely understand about the data throttling! And sorry to hear about the recent medical procedures. As I stated initially, I would be happy to compensate you for this work, something we can discuss over email should you be interested.

**Comment by GX4600** [comment]
Wait you own that site ? Holy cow I've been chasing my own tail in this conversstion. XD
Can the csv file be opened with this program ?
https://www.wps.com/en-US/

It's what i use for office documents and such. 
Feel free to send me the csv file and any other resources you feel could help in shaddoll constructing your latest goal. ;)
I'll do my best.
Sent from Yahoo Mail on Android 
 
  On Wed, Jul 10, 2019 at 7:47 AM, YGOPRODeck<notifications@github.com> wrote:   
Yes the Yu-Gi-Oh! Card Database is the site I own and run :P Thank you! I have personally found Card Rulings to be the absolute hardest piece of information to programmatically obtain which is why I am reaching out to someone who has more experience acquiring Card Rulings. I would love to add it to my card database.

I completely understand about the data throttling! And sorry to hear about the recent medical procedures. As I stated initially, I would be happy to compensate you for this work, something we can discuss over email should you be interested.

**Comment by AlanOC91** [comment]
Yep, that's me! :) 
I just downloaded that program and tested it and it seems to open my csv file without any issues so you should be ok with that!

Here is a link to the csv file: https://drive.google.com/open?id=1ZPQ3HoCFWpC-uXRJb_Tl5BPL8Va9A54i

It contains ID + Name of every card in my database. If you need more info then let me know!
Thank you! I really appreciate this! You've done awesome work here with this.

**Comment by chinhodado** [comment]
@AlanOC91 A few things

1) Sorry for the late reply, I don't log in to GitHub often recently and don't receive email notification about this.

2) I am the sole dev on this project. @GX4600 and @HoangDynasty are not affiliated with this project in any way and I have no idea who they are.

3) All the code for parsing wikia and wikipedia is here https://github.com/chinhodado/YgoSqliteDbCreator
Do with it as you wish.

**Comment by AlanOC91** [comment]
Wow this this has been incredibly strange after that revelation! Thank you for getting back to me and clearing things up. This has been beyond odd now!

I'll look into that right away, thank you!

**Comment by GX4600** [comment]
Lol, so am I still your underling for rulings production or has this made the tea a little bitter, lmk. ;)

Sent from Yahoo Mail on Android 
 
  On Sat, Jul 13, 2019 at 10:28 AM, YGOPRODeck<notifications@github.com> wrote:   
Wow this this has been incredibly strange after that revelation! Thank you for getting back to me and clearing things up. This has been beyond odd now!

I'll look into that right away, thank you!

**Comment by GX4600** [comment]
Hey guys,

Hope everyone is doing okay, I know this year has been quite rough on everyone. How are you guys, I've had at least 3 yu-gi-oh card shops close down permanently due to covid, I have maybe 2 solid card shops left. Haven't been out much or bought very little yugioh product this year due to restrictions. Lemme know how you guys are doing. 

Hopefully good. :)
